typewriter/fixes/fixer_utils.py

```diff
@@ -133,3 +145,221 @@ def touch_import(package, name, node):
+            # [Grammar] dotted_as_name: dotted_name ['as' NAME]
+            package = import_name = binding_name = None
+            name = node.children[0]
+            binding_name = str(node.children[2]).strip()
+            if name.type in (syms.dotted_name, token.NAME):
+                # [Grammar] dotted_name: NAME ('.' NAME)*
+                # We don't need the third field since we know the alias will be the binding_name
+                package, import_name, _ = decompose_name(name)
+            return package, import_name, binding_name
+        # As long as we aren't ignoring dotted names
+        # If there was no "as", we know this is dotted_name
+        # [Grammar] dotted_name: NAME ('.' NAME)*
+        return decompose_name(node)
+
+    package = None
+    imports = dict()  # type: Dict[str, Set[ImportPairing]]
+    if node.type == syms.import_name:
+        # [Grammar]: import_name: 'import' dotted_as_names
+        as_name = node.children[1]
+        if as_name.type in (syms.dotted_as_names, syms.import_as_names):
+            # [Grammar]: dotted_as_names: dotted_as_name (',' dotted_as_name)*
+            for child in as_name.children:
+                if str(child).strip() == ',':
+                    continue
+                package, import_name, binding_name = handle_name(child)
+                if package is None:
+                    package = 'no_pkg'
+                imports[package] = imports.get(package, set())
+                imports[package].add(ImportPairing(import_name, binding_name))
+        package, import_name, binding_name = handle_name(as_name)
+        if package is None:
+            package = 'no_pkg'
+        imports[package] = imports.get(package, set())
+        imports[package].add(ImportPairing(import_name, binding_name))
+    elif node.type == syms.import_from:
+        # [Grammar]:
+        # import_from: ('from' ('.'* dotted_name | '.'+)
+        #   'import' ('*' | '(' import_as_names ')' | import_as_names))
+        package = str(node.children[1]).strip()
+        if str(node.children[3]).strip() == '(':
+            # This means we are dealing with an import statement containing parentheses
+            # ex: from a import (b, c, d)
+            import_as_name = node.children[4]
+        else:
+            import_as_name = node.children[3]
+        if import_as_name.type == syms.import_as_names:
+            for child in import_as_name.children:
+                if str(child).strip() == ',':
+                    continue
+                _, import_name, binding_name = handle_name(child)
+                if package is None:
+                    package = 'no_pkg'
+                imports[package] = imports.get(package, set())
+                imports[package].add(ImportPairing(import_name, binding_name))
+        else:
+            _, import_name, binding_name = handle_name(import_as_name)
+            if package is None:
+                package = 'no_pkg'
```

Comment:
package is not set here, this should be done above... except `package` is never None, since it's the result of a `str.strip()`